sotodlib/io/imprinter.py

```diff
@@ -0,0 +1,672 @@
+import datetime as dt, os.path as op, os
+import numpy as np
+from collections import OrderedDict
+from typing import List
+
+import sqlalchemy as db
+from sqlalchemy import or_, and_, not_
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+
+from sotodlib.io.load_smurf import G3tSmurf, Observations as G3tObservations
+
+####################
+# useful constants #
+####################
+
+# book status
+UNBOUND = 0
+BOUND = 1
+FAILED = 2
+
+# tel tube, stream_id, slot mapping
+SLOTS = {
```

Comment:
this is the sort of thing that needs to be loaded from a config file and not hard coded